tools/tpm2_getekcertificate.c

```diff
@@ -605,30 +833,29 @@ static tool_rc process_output(void) {
      * the EK public hash in addition to the certificate data.
      * If so set the flag.
      */
-    if (ctx.rsa_cert_buffer) {
-        ctx.is_intc_cert = ctx.is_intc_cert ? ctx.is_intc_cert :
-        !(strncmp((const char *)ctx.rsa_cert_buffer,
+    bool is_intel_cert = ctx.manufacturer == VENDOR_INTEL;
+    if (!is_intel_cert && ctx.rsa_cert_buffer) {
+        is_intel_cert = !(strncmp((const char *)ctx.rsa_cert_buffer,
             "{\"pubhash", strlen("{\"pubhash")));
     }
 
-    if (ctx.ecc_cert_buffer) {
-        ctx.is_intc_cert = ctx.is_intc_cert ? ctx.is_intc_cert :
-        !(strncmp((const char *)ctx.ecc_cert_buffer,
+    if (!is_intel_cert && ctx.ecc_cert_buffer) {
+        is_intel_cert = !(strncmp((const char *)ctx.ecc_cert_buffer,
             "{\"pubhash", strlen("{\"pubhash")));
     }
 
     /*
      * Intel EK certificates on the NV-index are already in standard DER format.
      */
-    if (ctx.is_intc_cert && ctx.is_cert_on_nv) {
+    if (is_intel_cert && !ctx.is_cert_on_nv) {
```

Comment:
should not be changed, to prevent failures in integration tests. I think the complete if statement could be removed. But that could be done in later PR which implements correct usage of the raw option.